eacl/table.go

```diff
@@ -199,3 +200,60 @@ func (t *Table) UnmarshalJSON(data []byte) error {
 
 	return nil
 }
+
+// EqualTo compares Table between each other.
+func (t *Table) EqualTo(table *Table) bool {
```

Comment:
Can't it be a function? I assume we have access to all data through methods

Also lets define the operation only on values? IMO `nil` case is context-dependent and should be handled by the caller clearly.
```suggestion
func (t Table) EqualTo(table Table) bool {
```

Reply:
> Can't it be a function? I assume we have access to all data through methods

Yes, but performance is degraded (compare `BenchmarkTableEqualsComparison`  to `BenchmarkTableEqualToComparison`)

```
BenchmarkTableBinaryComparison
BenchmarkTableBinaryComparison-16        	  603984	      2328 ns/op
BenchmarkTableEqualToComparison
BenchmarkTableEqualToComparison-16       	20296022	        50.41 ns/op
BenchmarkTableEqualsComparison
BenchmarkTableEqualsComparison-16        	 9665785	       118.8 ns/op
BenchmarkTableBinaryComparison10
BenchmarkTableBinaryComparison10-16      	    9748	    104707 ns/op
BenchmarkTableEqualToComparison10
BenchmarkTableEqualToComparison10-16     	  498164	      2032 ns/op
BenchmarkTableEqualsComparison10
BenchmarkTableEqualsComparison10-16      	  283947	      3572 ns/op
BenchmarkTableBinaryComparison100
BenchmarkTableBinaryComparison100-16     	     138	   8734223 ns/op
BenchmarkTableEqualToComparison100
BenchmarkTableEqualToComparison100-16    	    4556	    266642 ns/op
BenchmarkTableEqualsComparison100
BenchmarkTableEqualsComparison100-16     	    2606	    386043 ns/op
```

Reply:
https://github.com/nspcc-dev/neofs-sdk-go/pull/180/commits/b15f32e45e9af720719f5dfad6ba7d08eeba1b09

Reply:
I got it. 

First of all, your implementation of `Equals` does not store slice getters in a variable (for-loops) which in general leads to performance loss.

Also as I see different approaches are used for `EqualTo` method and `Equals` function: the method uses private access to internal components (`Record`, `Filter`, `Target`) instead of getters, but a function does not. Although all of these types are in the same package, they are not internal to `Table` (imagine moving them to other packages).

So you can analyze performance in 2 ways:

1. Change `Equals` implementation: use `CID` and `Records` getters in `Equals` function, and use private access to its internals.
2. Change `EqualTo` implementation: access to `Table` internals using exported methods (getters).

IMO 2 s architecturally more appropriate (and we can replace method since there won't be a perfomance degradation), although 1 is also correct in the current packaging.

For example, here are results of 1 (`Alt` means changed implementation, others left untouched):
```
goos: linux
goarch: amd64
pkg: github.com/nspcc-dev/neofs-sdk-go/eacl/test
cpu: 11th Gen Intel(R) Core(TM) i5-11400F @ 2.60GHz
BenchmarkTableEqualToComparison-12         	52749079	        22.01 ns/op
BenchmarkTableEqualsComparison-12          	30736006	        37.40 ns/op
BenchmarkTableEqualsAltComparison-12       	49660705	        22.71 ns/op

BenchmarkTableEqualToComparison10-12       	  728594	      1570 ns/op
BenchmarkTableEqualsComparison10-12        	  555916	      2076 ns/op
BenchmarkTableEqualsAltComparison10-12     	 1000000	      1104 ns/op

BenchmarkTableEqualToComparison100-12      	    8246	    145467 ns/op
BenchmarkTableEqualsComparison100-12       	    6166	    196389 ns/op
BenchmarkTableEqualsAltComparison100-12    	   10000	    107600 ns/op
```